be/src/exprs/agg/count_if.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "column/nullable_column.h"
+#include "column/type_traits.h"
+#include "column/vectorized_fwd.h"
+#include "exprs/agg/aggregate.h"
+#include "gutil/casts.h"
+#include "util/value_generator.h"
+
+namespace starrocks {
+
+struct CountIfFunctionState {
+    int64_t count = 0;
+};
+
+// count not null column
+template <LogicalType LT>
+class CountIfAggregateFunction final
+        : public AggregateFunctionBatchHelper<CountIfFunctionState, CountIfAggregateFunction<LT>> {
+public:
+    using InputColumnType = RunTimeColumnType<LT>;
+
+    template <bool is_add>
+    void do_update(FunctionContext* ctx, const Column** columns, AggDataPtr __restrict state, size_t row_num) const {
+        const auto& column = down_cast<const InputColumnType&>(*columns[0]);
+        if (column.get_data()[row_num]) {
```

Comment:
Do you mean `if constexpr`? If so, it will only be judged at compile time and will not have an impact on runtime performance

Reply:
replace with branchless
```
if constexpr (is_add) {
  this->data(state).count += column.get_data()[row_num];
} else {
   this->data(state).count -= column.get_data()[row_num];
}
```

Reply:
I thought about this too, but eventually didn't do it because the data in the column may not be of type bool in some cases.
For example, executing select if(3.14, 1, 0) within sr or MySql all results in 1, because 3.14 is a non-zero value and is therefore judged to be true, which I keep consistent in count_if, so this branch cannot be simplified away like this

Reply:
It seems easier to cause confusion, and probably what you really want to achieve is count_if(bool). Direct support for count_if doesn't seem to make practical sense